### ISSUE TITLE
Heteroscedastic Multiclass Classification?

### ISSUE THREAD

**Issue by austinmcdannald** [GPflow/GPflow#1531]
I am trying to do multiclass classification with a training data set that has the uncertainty of class membership for each class for each point. I would like to propagate these uncertainties through the GPC in a similar manner to what can be done with Heteroscedastic Regression.

My thought is that this would require some modification to the built-in GPflow Classes. Multiclass GPC uses the  `gpflow.likelihoods.multiclass` module, which (indirectly) contains the `gpflow.likelihoods.RobustMax` class.  These classes would likely have to be modified wherever `epsilon` and `eps_k1` are mentioned. 

Does anyone have ideas on how best to modify these classes to allow for heterscedastic multiclass classification?

**Comment by st--** [comment]
Hi @austinmcdannald, your question isn't clear enough by itself to be able to give you a meaningful answer. In classification, the squashed outputs already correspond to probabilities for the different classes. Note that the Softmax classification model generally seems to be better behaved (though it requires you to do stochastic optimization). Feel free to re-open this issue with more detail, and/or join us on the GPflow slack ([link in the README.md](github.com/GPflow/GPflow/#slack-workspace)) if you'd like to discuss this further :)
